### ISSUE TITLE
WindowState of ShellWindow... ViewModel communication...

### ISSUE THREAD

**Issue by Aitania** [WildGums/Orchestra#50]
Hallo Geert,

1) is a way to change the size of the ShellWindow at startup?
            var serviceLocator = ServiceLocator.Default;
            var shellService = serviceLocator.ResolveType<IShellService>();
            await shellService.CreateWithSplashAsync<ShellWindow>();
// I want to do sommething like this:
            ShellWindow.WindowStateProperty = WindowState.Normal; 

2) I have notice that the whole application is divided into several windows (catel:UserControls)
   a) The RibbonView b) the MainView c) the StatusBarView
       Actually i want to have the Syncfusion Ribbon. 
 Now i have this scenario: 
 I have a command triggered from a backstage fluent RibbonButton. 
 The command is captured in RibbonViewModel.
  RibbonViewModel send it further to "existing" MainViewModel.
  I need the vm of the MainViewModel which contains the binding property of the MainView.
  The MainView is to display the data which are filled in a List according to Model which happens in a  
  extra  class
  I take the existing instance of the MainViewModel like this:

```
    /// <summary>
    /// Method to invoke when the OpenProject command is executed.
    /// </summary>
    private async void OnNewExecute()
    {
        try
        {
            var viewManager = ServiceLocator.Default.ResolveType<IViewManager>();
            foreach (var view in viewManager.ActiveViews)
            {
                var vm = view.ViewModel;
                if (vm != null)
                {
                    if (vm.GetType() == typeof(MainViewModel))
                    {
                        mainViewModel = vm as MainViewModel;
                        mainViewModel.DisplayCatalog();
                        break;
                    }
                }
            }
        }
        catch (Exception e)
        {
            Log.Error(e.Message);
        }
    }

  Is it possible to get the mainViewModel thru Catel?
```

Best Regards,
Michael

**Comment by GeertvanHorrik** [comment]
1. Not at the moment, we don't expose this via an api. However... you can retrieve the window in the shell (there is a "AfterCreatingShell"-something method) and set the value there.

2. We recommend not to mess around with view models when using commands. The way to implement this is by command containers. These are simple & small containers that implement a command and are registered in the service locator. This way, your commands don't live on view models but are self-managed. You can then use the *CommandManagerBinding* in Catel to bind to such a command. I think the examples even show how to use this.

Basically, the view models represent a state for a view. You shouldn't be requesting the information from a VM, but from a service (which survives the view / view models life cycle and can remain state).

**Comment by Aitania** [comment]
Hallo Geert,
1. You are right. I solved it in the method:
        _public override async Task InitializeAfterCreatingShellAsync()
        {
            Log.Info("Delay to show the splash _screen");_
            _var application = System.Windows.Application.Current;_
            ___application.MainWindow.WindowState = System.Windows.WindowState.Normal;__ 
}
2. 
"We recommend not to mess around with view models when using commands. "

"This way, your commands don't live on view models but are self-managed."
Yes.. I know that is not MVVM conform. 

I do use the             
_var commandManager = ServiceLocator.Default.ResolveType<ICommandManager>();
var commandInfoService = ServiceLocator.Default.ResolveType<ICommandInfoService>();_

But my binding and my controls are based on the view/vm.. If i execute the command e.g. in the ApplicationInitializationService class how controls are get the data?
What if the command is triggered in ribbon and the databinding is in the MainViewWindow...?

Best Regards,
Michael

**Comment by GeertvanHorrik** [comment]
In such a case, we would have:

1. An independent service that manages the current data
2. An independent command container that can interact with this data
3. MainViewModel exposes this data so it can be shown in the view (but the data is coming from the service that is available for both 2 and 3).
4. The ribbon can invoke the command because it's an application command, not a view-related command

If you really only have this data available in the main vm for some reason, you should still create a command container and call AddAction in the vm.InitializeAsync and RemoveAction in the vm.CloseAsync.

**Comment by GeertvanHorrik** [comment]
Closing this issue. Let us know if you need more info.
